File: Framework/src/main/java/org/example/loader/PageRegistry.java
package org.example.loader;

import org.example.core.event.IEvent;
import org.example.core.loader.IPageRegistry;
import org.example.core.page.presentation.AbstractIPage;

import java.util.HashMap;
import java.util.Map;

class PageRegistry implements IPageRegistry {
    private final Map<IEvent, AbstractIPage> pages = new HashMap<>();

    @Override
    public void register(IEvent event, AbstractIPage page) {
        pages.put(event, page);
    }

    @Override
    public AbstractIPage get(IEvent event) {
        return pages.get(event);
    }
}

File: framework/src/main/kotlin/org/example/loader/PageRegistry.kt
package org.example.loader

import org.example.core.intent.IIntent
import org.example.core.page.presentation.AbstractIPage
import org.example.core.registry.IPageRegistry

internal class PageRegistry: IPageRegistry {
    private val pages: MutableMap<IIntent?, AbstractIPage?> = HashMap()

    override fun register(
        intent: IIntent?,
        page: AbstractIPage?
    ) {
        pages[intent] = page
    }

    override fun get(intent: IIntent?): AbstractIPage? {
        return pages[intent]
    }
}